feature-libs/product-configurator/rulebased/occ/variant/converters/occ-configurator-variant-normalizer.spec.ts

```diff
@@ -399,6 +407,27 @@ describe('OccConfiguratorVariantNormalizer', () => {
         targetFromPredecessor.interactionState
       );
     });
+
+    it('should convert a configuration with kb key', () => {
+      const result = occConfiguratorVariantNormalizer.convert(configuration);
+      expect(result.kbKey).toBeDefined();
+      expect(result.kbKey.kbName).toEqual(configuration.kbKey.kbName);
+      expect(result.kbKey.kbLogsys).toEqual(configuration.kbKey.kbLogsys);
+      expect(result.kbKey.kbVersion).toEqual(configuration.kbKey.kbVersion);
+      expect(result.kbKey.kbBuildNumber).toEqual(
+        configuration.kbKey.kbBuildNumber
+      );
+    });
+
+    it('should convert a configuration with undefined kb key', () => {
+      configuration.kbKey = {};
+      const result = occConfiguratorVariantNormalizer.convert(configuration);
+      expect(result.kbKey).toBeDefined();
```

Comment:
does it really make sense to add a kbkey object with undefined fields to the configuration? - i would expect that kbkey itself is undefined, if no kbkey is provided.

Reply:
I extended configuration with corresponding kbKex data: 
![image](https://user-images.githubusercontent.com/61147963/184622279-4e86b0df-b7b2-4be4-8b4f-f8250ac9a7ae.png)
That's why I set configuration.kbKey to empty object

Reply:
What i meant is:
your test expects that, if the kbKey is undefined in the OCC configuration, that it is still defined in the model after converting, but with all fields being undefined. But why - do we have any specific use case for such a behaviour?
I would expect, if the kbKey is undefined in the occ configuration, that it is also undefined in the model, after converting. In the model the kbKey is marked as optional, so it is legit to set it to undefined.

Reply:
Done!